packages/renderer/src/lib/ui/Button.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+import type { ButtonType } from './Button';
+import Spinner from './Spinner.svelte';
+
+export let title: string = undefined;
+export let inProgress = false;
+export let disabled = false;
+export let type: ButtonType = undefined;
+
+let classes = '';
+$: {
+  if (disabled || inProgress) {
+    if (type === 'primary' || type === 'secondary') {
+      classes = 'bg-charcoal-50';
+    } else {
+      classes = 'text-charcoal-50 no-underline';
+    }
+  } else {
+    if (type === 'primary') {
+      classes = 'bg-purple-500 border-none hover:bg-purple-400';
+    } else if (type === 'secondary') {
+      classes = 'border-[1px] border-gray-200 hover:border-purple-500 hover:text-purple-500';
+    } else {
+      classes = 'border-none hover:underline';
```

Comment:
it makes me think, do we need `<button>` or `<primary-button>` `<secondary-button>` etc

Reply:
Since most are primary, I think it'll be fine if we make that the default.